httpx/dispatch/http2.py

```diff
@@ -139,16 +142,14 @@ def initiate_connection(self) -> None:
             # read timeouts, if we haven't already.
             self.timeout_flags[stream_id].set_read_timeouts()
             if isinstance(event, h2.events.ResponseReceived):
-                break
-
-        status_code = 200
-        headers = []
-        for k, v in event.headers:
-            if k == b":status":
-                status_code = int(v.decode("ascii", errors="ignore"))
-            elif not k.startswith(b":"):
-                headers.append((k, v))
-        return status_code, headers
+                status_code = 200
```

Comment:
I made this change because it helped me understand what the code was doing, it is functionally the same as the code before the change. I am fine with whichever version you prefer.

Reply:
I think I prefer the earlier version because the exit point isn't in the `while` loop. I'm going to make the change and then this PR looks good to me. :)